File: backend/satisfactory-blueprint/satisfactory-blueprint/src/main/java/com/satisfactory/blueprint/dto/CreateTransportPlanRequest.java
package com.satisfactory.blueprint.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CreateTransportPlanRequest {
    private String name;
    private Long plannerId;
    private boolean addAllResources;
    private List<TransportItemRequestDto> transportItems;
}

File: frontend/satisfactory-blueprint-designer/src/types/createTransportPlanRequest.ts
// src/types/createTransportPlanRequest.ts
import type { TransportItemRequestDto } from './transportItemRequest';

/**
 * Mirrors com.satisfactory.blueprint.dto.CreateTransportPlanRequest
 */
export interface CreateTransportPlanRequest {
  name: string;
  plannerId: number;
  addAllResources: boolean;
  transportItems: TransportItemRequestDto[];
}
